File: 2023/template.ex
defmodule DayNZ_DAY do
  def run(mode) do
    data = read_input(mode)

    [{1, data}, {2, data}]
    |> Task.async_stream(
      fn
        {1, data} -> {1, data |> part1}
        {2, data} -> {2, data |> part2}
      end,
      timeout: :infinity
    )
    |> Enum.reduce({0, 0}, fn
      {_, {1, res}}, {_, p2} -> {res, p2}
      {_, {2, res}}, {p1, _} -> {p1, res}
    end)
  end

  def read_input(:test) do
    "{test-data}"
    |> String.split("\n")
    |> prepare_data
  end

  def read_input(:actual) do
    File.stream!("input-THE_DAY")
    |> Enum.map(fn n -> n |> String.trim() end)
    |> prepare_data
  end

  def prepare_data(data) do
    data
  end

  def part1(data) do
    1
  end

  def part2(data) do
    2
  end
end

File: 2024/template.py

def run(data):
    data = parse_data(data)
    return (part1(data), part2(data))

def parse_data(data):
    return data

def part1(data):
    return 0

def part2(data):
    return 0


def test():
    test_data = """
_SOME_NONSENSE_
"""
    assert part1(parse_data(test_data)) == 11, "Should be 11"
    assert part2(parse_data(test_data)) == 31, "Should be 31"
